lib/src/main/java/com/auth0/jwt/jwts/JWT.java

```diff
@@ -159,43 +219,43 @@ static Verification init(Algorithm algorithm) throws IllegalArgumentException {
 
         @Override
         public Verification withNbf(long nbf) {
-            throw new UnsupportedOperationException("you shouldn't be calling this method");
+            throw new UnsupportedOperationException("this method has not been implemented");
```

Comment:
Why is it that we have all these methods that shouldn't be called?

Reply:
because they're implementing the interface method.  they are required to implement it or declare the class abstract.

Reply:
@lbalmaceda

Reply:
The comment from @lyzhang27 is correct. You should **not** expose methods you don't want your users to call. A good inheritance design (parent/child classes) can prevent this.